docs/integrations/pytorch_lightning.rst

```diff
@@ -1,318 +1,305 @@
-            x, y = batch
-            y_hat = self.forward(x)
-            loss = F.cross_entropy(y_hat, y)
-            tensorboard_logs = {'train_loss': loss}
-            return {'loss': loss, 'log': tensorboard_logs}
+Log test metrics
+^^^^^^^^^^^^^^^^
 
-        def validation_step(self, batch, batch_idx):
-            # OPTIONAL
-            x, y = batch
-            y_hat = self.forward(x)
-            return {'val_loss': F.cross_entropy(y_hat, y)}
+Log additional metrics
+^^^^^^^^^^^^^^^^^^^^^^
 
-        def validation_end(self, outputs):
-            # OPTIONAL
-            avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
-            tensorboard_logs = {'val_loss': avg_loss}
-            return {'avg_val_loss': avg_loss, 'log': tensorboard_logs}
+Log performance charts
+^^^^^^^^^^^^^^^^^^^^^^
 
-        def test_step(self, batch, batch_idx):
-            # OPTIONAL
-            x, y = batch
-            y_hat = self.forward(x)
-            return {'test_loss': F.cross_entropy(y_hat, y)}
+Save model checkpoints
+^^^^^^^^^^^^^^^^^^^^^^
 
-        def test_end(self, outputs):
-            # OPTIONAL
-            avg_loss = torch.stack([x['test_loss'] for x in outputs]).mean()
-            tensorboard_logs = {'test_loss': avg_loss}
-            return {'avg_test_loss': avg_loss, 'log': tensorboard_logs}
+Troubleshooting
+---------------
 
-        def configure_optimizers(self):
-            # REQUIRED
-            # can return multiple optimizers and learning_rate schedulers
-            # (LBFGS it is automatically supported, no need for closure function)
-            return torch.optim.Adam(self.parameters(), lr=LR)
+How to ask for help?
+^^^^^^^^^^^^^^^^^^^^
+The fastest way is to simply chat with us. Chat icon is located directly in-app, in the lower right corner. Use it!
 
-        @pl.data_loader
-        def train_dataloader(self):
-            # REQUIRED
-            return DataLoader(MNIST(os.getcwd(), train=True, download=True, transform=transforms.ToTensor()), batch_size=BATCHSIZE)
+.. image:: ../_static/images/integrations/chat-icon.png
+   :target: ../_static/images/integrations/chat-icon.png
+   :alt: Chat icon
 
-        @pl.data_loader
-        def val_dataloader(self):
-            # OPTIONAL
-            return DataLoader(MNIST(os.getcwd(), train=True, download=True, transform=transforms.ToTensor()), batch_size=BATCHSIZE)
+For more general questions go to our |forum|.
 
-        @pl.data_loader
-        def test_dataloader(self):
-            # OPTIONAL
-            return DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=transforms.ToTensor()), batch_size=BATCHSIZE)
+Other integrations you may like
+-------------------------------
+Here are other integrations with libraries from the PyTorch ecosystem:
 
+#. |PyTorch|
+#. |PyTorch Ignite|
+#. |Catalyst|
+#. |skorch|
 
-    from pytorch_lightning.loggers.neptune import NeptuneLogger
+You may also like these two integrations:
```

Comment:
I don't think those are related -> I'd rather show optuna or something

Reply:
I see. I put optuna and plotly.